acceptance/tests/provider/package/dpkg_ensure_held_package_is_latest.rb

```diff
@@ -0,0 +1,26 @@
+test_name "puppet provider package dpkg"
+confine :to, :platform => /debian-8-amd64/
+require 'puppet/acceptance/common_utils'
+extend Puppet::Acceptance::PackageUtils
+extend Puppet::Acceptance::ManifestUtils
+
+
+package = "nginx"
+
+agents.each do |agent|
+  teardown do
+    package_absent(agent,package, '--force-yes')
+  end
+end
+
+step"Ensure that package is installed first if not present" do
+  info = on(agent.name, "apt-cache policy #{package} | grep Candidate:").stdout
```

Comment:
info and info2 are not very sugestive, also you could define define a methods that reads package information like:
```
def package_info(agent, package)
  on(agent.name, "apt-cache policy #{package} | grep Candidate:").stdout.split
end
```